### PR TITLE
FormRecognizer: Updating Train API to take optional parameter.

### PR DESCRIPTION
**Pull request by ramparab**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by ramparab** [comment]
@admins: Please let me know if you need any additional information to complete the review and the PR.

**Comment by ramparab** [comment]
@Admins; please let me know if there is anything I can do to unblock you.

**Comment by jhendrixMSFT** [comment]
@ramparab does anybody else from your team need to review this?  I see code owners have been tagged for review.

**Comment by ramparab** [comment]
> I see code owners have been tagged for review.
@jhendrixMSFT: Hello, where can I find who from my team is tagged?

**Comment by jhendrixMSFT** [comment]
Top-right of the PR under `Reviewers` column, the people with the lock next to their aliases.

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json
Comment (yangyuan) [inline specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json]: Should update samples to show the usage.

**Review by yangyuan — COMMENTED** [review]
This change is a design change, which requires a review from Azure API Review Board.
Please schedule a review with the Review Board (online or offline).

File: specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json
Comment (ramparab) [reply specification/cognitiveservices/data-plane/FormRecognizer/preview/v1.0/FormRecognizer.json]: Done.

**Review by krishnand — COMMENTED** [review]
1. Can we remove lines 38-49.. we just need to show that documents used in training are reported
2. Line 320 -- A flag to indicate **if** ..

**Comment by ramparab** [comment]
@Krishnand: 
1. re: TrainWithSubFolders.json: I want to keep this example same as Train.json to show optional parameters with defaults, hence these files are same.
2. re: FormRecognizer.json: done.

**Review by yangyuan — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**
